File: tcbn/ruby/update/tcbn_update.rb
#! /usr/bin/ruby
# -*- coding: utf-8 -*-
#
#	ruby/update/tcbn_update.rb
#
#					Mar/22/2013
#
# -------------------------------------------------------------
require 'date'
require 'tokyocabinet'
include TokyoCabinet
require 'json'
#
load '/var/www/data_base/common/ruby_common/tcbn_manipulate.rb'
#
# -------------------------------------------------------------
puts	"*** 開始 ***"
id_in = ARGV[0]
population_in = ARGV[1].to_i
#
puts id_in,population_in
#
bdb = BDB::new
#
file_in="/var/tmp/tokyo_cabinet/cities.tcb"
if !bdb.open(file_in, BDB::OWRITER | BDB::OCREAT)
	ecode = bdb.ecode
	STDERR.printf("open error: %s\n", bdb.errmsg(ecode))
end
#
tcbn_update_proc(bdb,id_in,population_in)
#
tcbn_display_proc(bdb)
#
# close the database
if !bdb.close
	ecode = bdb.ecode
	STDERR.printf("close error: %s\n", bdb.errmsg(ecode))
end
#
puts "*** 終了 ***"
# -------------------------------------------------------------

File: tcbn/perl/update/tcbn_update.pl
#! /usr/bin/perl
#
#	tcbn_update.pl
#
#					Sep/21/2010
# ----------------------------------------------------------
use	TokyoCabinet;
use	strict;
use	warnings;
use	Encode;
use	JSON;
use	utf8;
#
use	lib '/var/www/data_base/common/perl_common';
use	kvalue_manipulate;
use	text_manipulate;
use	tcbn_manipulate;
# ----------------------------------------------------------
print (encode ('utf-8',"*** 開始 ***\n"));

my $file_in = $ARGV[0];
my $id_in = $ARGV[1];
my $population_in = $ARGV[2];

print ("$id_in\t$population_in\n");
my $bdb = TokyoCabinet::BDB->new();

if(!$bdb->open($file_in, $bdb->OWRITER | $bdb->OCREAT))
{
	my $ecode = $bdb->ecode();
	printf STDERR ("open error: %s\n", $bdb->errmsg($ecode));
}


tcbn_manipulate::data_update_proc ($bdb,$id_in,$population_in);

# traverse records
my $cur = TokyoCabinet::BDBCUR->new($bdb);
$cur->first();
while(defined(my $key = $cur->key()))
	{
	my $json_str = $cur->val();
	if(defined($json_str))
		{
		kvalue_manipulate::json_display_proc ($key,$json_str);
		}
	$cur->next();
	}

# close the database
if(!$bdb->close()){
	my $ecode = $bdb->ecode();
	printf STDERR ("close error: %s\n", $bdb->errmsg($ecode));
}

print (encode ('utf-8',"*** 終了 ***\n"));

# ----------------------------------------------------------
